src/librustc/lint/builtin.rs

```diff
@@ -224,6 +224,12 @@ declare_lint! {
 }
 
 declare_lint! {
+    pub LEGACY_CONSTRUCTOR_VISIBILITY,
+    Deny,
```

Comment:
3 is small, no?
I preferred deny-by-default mostly to avoid successfully compiling more illegal code by default.